BroadcomCloudSwg/connectors/broadcom_cloud_swg_connector.py

```diff
@@ -0,0 +1,197 @@
+            return self._broadcom_cloud_swg_client
+
+        self._broadcom_cloud_swg_client = BroadcomCloudSwgClient(
+            username=self.module.configuration.username,
+            password=self.module.configuration.password,
+            rate_limiter=self.rate_limiter(),
+        )
+
+        return self._broadcom_cloud_swg_client
+
+    async def get_events(self) -> list[str]:
+        """
+        Collects events from platform and push to intakes.
+
+        Returns:
+            list[str]:
+        """
+        end_date = datetime.utcnow()
+        start_date = self.last_event_date
+
+        token = None
+        continue_processing = True
+
+        data_to_push: list[dict[str, str]] = []
+        result: list[str] = []
+
+        while continue_processing:
+            continue_processing, token, file_name = await self.broadcom_cloud_swg_client.get_report_sync(
+                start_date=start_date,
+                end_date=end_date,
+                token=token,
+            )
+
+            async with aiofiles.open(file_name) as file:
+                headers = None
+                async for line in file:
+                    if line.startswith("#"):
+                        headers = BroadcomCloudSwgClient.parse_string_as_headers(line)
+                    else:
+                        data_to_push.append(BroadcomCloudSwgClient.parse_input_string(line, headers))
+
+                        data_to_push = BroadcomCloudSwgClient.reduce_list(data_to_push)
+
+                        if len(data_to_push) >= self.configuration.chunk_size:
+                            result.extend(
+                                await self.push_data_to_intakes(
+                                    [orjson.dumps(event).decode("utf-8") for event in data_to_push]
+                                )
+                            )
+
+                            data_to_push = []
+
+            result.extend(
+                await self.push_data_to_intakes([orjson.dumps(event).decode("utf-8") for event in data_to_push])
+            )
+            data_to_push = []
+
+            await delete_file(file_name)
+
+        with self.context as cache:
+            logger.info(
+                "New last event date now is {last_event_date}",
+                last_event_date=end_date.isoformat(),
+            )
+
+            cache["last_event_date"] = end_date.isoformat()
+
+        return result
+
+    def run(self) -> None:  # pragma: no cover
+        """Runs BroadcomCloudSwgConnector."""
+        while self.running:
+            try:
+                loop = asyncio.get_event_loop()
+
+                previous_processing_end = None
+
+                while self.running:
+                    processing_start = time.time()
+                    if previous_processing_end is not None:
+                        EVENTS_LAG.labels(intake_key=self.configuration.intake_key).observe(
+                            processing_start - previous_processing_end
```

Comment:
To have the correct lag, the difference must apply between the `processing_start` and `last_event_date`.